### ISSUE TITLE
mpd-0.21:  undefined reference to symbol 'ogg_sync_clear'

### ISSUE THREAD

**Issue by Zlogene** [MusicPlayerDaemon/MPD#400]
Can not link against ogg re-writing gentoo ebuild, but the ogg header is in place.

/usr/lib/gcc/x86_64-pc-linux-gnu/7.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: src/decoder/plugins/libdecoder_plugins.a(VorbisDecoderPlugin.cxx.o): undefined reference to symbol 'ogg_sync_clear'
/usr/lib64//libogg.so.0: error adding symbols: DSO missing from command line

**Comment by MaxKellermann** [comment]
Please paste the linker command

**Comment by Zlogene** [comment]
I actually found the root of the evil bumping gentoo ebuild, almost every single decoder plugin require `-Dflac` to be enabled (and therefore pull the flac library in it). Initially I came up with the following solution (which means enable `-Dvorbis` if `-Dflac` enabled, enable `-Dflac` and `-Dvorbis` if `-Dmpg123` enabled and so on):

``` bash
REQUIRED_USE="
    flac? ( vorbis )
    mpg123? ( flac vorbis )
    opus? ( flac )
    sid? ( flac )
    sndfile? ( flac )
    vorbis? ( flac )
" 
```

because flac fails to link without vorbis and vorbis without flac... But I gave up on providing conditions and rather enabled `-Dflac` and `-Dvorbis` for users by default as `+flac` and `+vorbis` can be seen here  https://gitweb.gentoo.org/repo/gentoo.git/tree/media-sound/mpd/mpd-0.21.ebuild#n18.
I do not know if this is expected behaviour or not, but would be nice to make the buildsystem die on this conditon, like `upnp` dies without `expat` being set.
